### PR TITLE
(Backend) Combine BYON image logic and normal imagestream logic

### PR DESCRIPTION
**Pull request by maroroman**
This PR attempts to combine both BYON images backend and the backend that was used in the old spawner UI.

Main changes: 
The endpoint for both BYON and Imagestreams is now /api/images/:type, for now for BYON the type value should be "byon". For notebook-image labeled imagestreams any other ImageType can be put in (A.K.A. ImageType = 'byon' | 'jupyter' | 'other')
All references to notebooks and notebook iamges have been changed to images and BYON images for clarity. This has been done in frontend as well.
The POST, PUT and DELETE calls work only for BYON images currently, we could add a labels section in the request body to set up what gets created.

All other stuff is mostly the same as PR #197

Resolves: #252

### CODE REVIEW

File: frontend/src/pages/notebookImages/NotebookImages.tsx [hunk 1]
```diff
@@ -23,8 +23,8 @@ const description = `Import, delete, and modify notebook images.`;
 const NotebookImages: React.FC = () => {
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/notebookImages/NotebookImages.tsx:23]: Can you also rename the component, file name, and repo name to BYONImages? Also, could you update the other files in this repo?

**Review by DaoDaoNoCode — CHANGES_REQUESTED** [review]
Overall lgtm! Left some comments to address, and could you also remove the comments in the code?

File: backend/src/types.ts [hunk 2]
```diff
@@ -426,3 +404,83 @@ export type PipelineRunKind = {
...
+
+export type ImageInfo = {
+  name: string;
+  tags: ImageTagInfo[];
+  description?: string;
+  url?: string;
+  display_name?: string;
+  default?: boolean;
+  order?: number;
+};
```
[the remark this answers is not in the document]
Comment (maroroman) [reply backend/src/types.ts:486]: adding `export type ImageType = "byon" | "jupyter" | "other";`

**Review by DaoDaoNoCode — APPROVED** [review]
Tested, works well, lgtm! Could you squash the commits?

**Comment by maroroman** [comment]
> Tested, works well, lgtm! Could you squash the commits?

We can do squash on merge, I think it is easier

**Review by lucferbux — APPROVED** [review]
/lgtm

**Comment by lucferbux** [comment]
Tested and works great!

**Comment by lucferbux** [comment]
> > Tested, works well, lgtm! Could you squash the commits?
> 
> We can do squash on merge, I think it is easier

Yeah, we can totally do that, although it would be super nice if we can make it into one squashed commit signed off with your gpg key, just to get the git History ordered.

File: backend/src/routes/api/images/imageUtils.ts [hunk 3]
```diff
@@ -1,17 +1,142 @@
...
+  TagContent,
   KubeFastifyInstance,
-  NotebookImage,
-  ImageStreamListKind,
-  ImageStreamKind,
-  NotebookImageStatus,
-  NotebookImageCreateRequest,
-  NotebookImageUpdateRequest,
-  NotebookImagePackage,
+  ImageStreamList,
```
[the remark this answers is not in the document]
Comment (maroroman) [reply backend/src/routes/api/images/imageUtils.ts]: Resolved all related to Images, but will resolve ones related to notebooks in a refactoring PR later so its in the correct PR.

**Comment by maroroman** [comment]
Rebased and squashed @lucferbux  @DaoDaoNoCode  @dlabaj  What should be done about the uncooperative tests?

File: package.json [hunk 4]
```diff
@@ -23,7 +23,7 @@
     "dev": "run-p -l dev:*",
     "dev:backend": "cd ./backend && npm run start:dev",
     "dev:frontend": "cd ./frontend && npm run start:dev",
-    "format": "prettier --write \"backend/**/*.js\" \"frontend/**/*.ts\" \"frontend/**/*.tsx\"",
+    "format": "prettier --write \"backend/**/*.ts\" \"frontend/**/*.ts\" \"frontend/**/*.tsx\"",
```
[the remark this answers is not in the document]
Comment (maroroman) [reply package.json]: It seems that didn't help sadly

**Pull request merged**